### PR TITLE
[WIP]infra function to create Ubuntu OS

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it

sealer supports centOS and ubuntuOS when creating a cluster

### Describe how you did it

Add imageId field in clusterfile：
ex: 
apiVersion: zlink.aliyun.com/v1alpha1
kind: Cluster
metadata:
  name: my-test-cluster
spec:
  image: sealer-io/kubernetes:v1.19.8
  provider: ALI_CLOUD
  ssh:
    passwd: Sealer123
    pk: xxx
    pkPasswd: xxx
    user: root
  network:
    podCIDR: 100.64.0.0/10
    svcCIDR: 10.96.0.0/22
  certSANS:
    - aliyun-inc.com
    - 10.0.0.2

  masters:
    cpu: 4
    memory: 4
    count: 1
    imageId: Ubuntu/CentOS
    systemDisk: 100
    dataDisks:
      - 100
  nodes:
    cpu: 4
    memory: 4
    count: 1
    imageId: Ubuntu/CentOS
    systemDisk: 100
    dataDisks:
      - 100

### CODE REVIEW

File: pkg/infra/aliyun/ali_ecs.go [hunk 1]
```diff
@@ -445,7 +445,12 @@ func (a *AliProvider) RunInstances(instanceRole string, count int) error {
 
 	request := ecs.CreateRunInstancesRequest()
 	request.Scheme = Scheme
-	request.ImageId = ImageID
+	system := a.Cluster.Spec.Masters.ImageID
+	if system == CentOS || system == "" {
```
Comment (kakaZhou719) [inline pkg/infra/aliyun/ali_ecs.go:449]: simple the code.

File: pkg/infra/aliyun/ali_ecs.go [hunk 2]
```diff
@@ -445,7 +445,12 @@ func (a *AliProvider) RunInstances(instanceRole string, count int) error {
...
+		request.ImageId = CentOSImageID
+	} else if system == Ubuntu {
```
Comment (bxy4543) [inline pkg/infra/aliyun/ali_ecs.go:451]: what happens in other situations? All situations should be considered at all times.

File: pkg/infra/aliyun/ali_ecs.go (hunk 1 above)
Comment (bxy4543) [inline pkg/infra/aliyun/ali_ecs.go:449]: why cluster ImageID is centos ?
this is no longer the meaning of `ID`.

File: types/api/v1/cluster_types.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ type Hosts struct {
 	CPU        string   `json:"cpu,omitempty"`
 	Memory     string   `json:"memory,omitempty"`
 	Count      string   `json:"count,omitempty"`
+	ImageID    string   `json:"imageId,omitempty"`
```
Comment (allencloud) [inline types/api/v1/cluster_types.go:39]: Infra package is quite decoupled with the sealer main logic. Then when hacking infra package, we should not modify anything from sealer main package.

Will we take vm image into consideration? This should be discussed, too. Now I disagree with that.

Comment (Stevent-fei) [reply types/api/v1/cluster_types.go:39]: ok, thanks for your valuable comments，maybe there is a better way to solve this problem.

**Pull request closed without merging**